rdrf/rdrf/views/registration_rdrf.py

```diff
@@ -26,6 +27,7 @@ def load_registration_class(self, request, form):
             registration_class = import_string(settings.REGISTRATION_CLASS)
             return registration_class(request, form)
 
+    @csp_update(DEFAULT_SRC=("https://www.google.com", "https://www.gstatic.com"))
```

Comment:
The other way would be to add nonces to all scripts, but I think that'd be a tricky change because we are injecting some scripts in the code.

Reply:
Also, we could try and make these exceptions more specific, so that only scripts from google are accepted, not stylesheets etc.

There is no 'inheritance' in CSP (setting a script policy means that all default policy settings are ignored), so we'd need to add a script-specific policy that's a copy of the default one, then update that script policy for this view.

The likelihood of being compromised by something other than a script on google's static domains seems extremely low, but maybe we'll want to include other 3rd party scripts from less trustworthy sources in the future. The security auditors may have suggestions about this during the re-test.

Reply:
I think there's merit into investigating into this a bit as without nonces having `unsafe-inline` it's not offering much protection. It might be tricky as you said due to injecting some JS scripts in the code (some widgets, DSL evaluation code) but I wonder if we can use the nonce to set it to the scripts we're injecting.

Reply:
Good point. They have recommendations here: https://docs.newrelic.com/docs/browser/new-relic-browser/getting-started/compatibility-requirements-new-relic-browser#csp

So what do you think about splitting up the policies for scripts, fonts, objects etc. @alintreznai @sztamas?
I think having code duplication for each policy isn't so bad as long as we can keep it clean-looking. We might be adding more external scripts in the future so I think we should be proactive in getting this right and being as locked-down as possible.